app/Http/Controllers/ConferencesController.php

```diff
@@ -47,28 +47,40 @@ public function index(Request $request)
 
         switch ($request->input('sort')) {
             case 'date':
-                $conferences = $conferences->sortBy(function (Conference $model) {
-                    return $model->starts_at;
+                $conferences = $conferences->sortBy(function ($conference) {
+                    if ($conference->starts_at->isPast()) {
+                        return $conference->starts_at->addCentury();
+                    }
+
+                    return $conference->starts_at;
                 });
                 break;
-            case 'closing_next':
-                // Forces closed CFPs to the end. I feel dirty. Even dirtier with the 500 thing.
-                $conferences = $conferences
-                    ->sortBy(function (Conference $model) {
-                        if ($model->cfp_ends_at > Carbon::now()) {
-                            return $model->cfp_ends_at;
-                        } elseif ($model->cfp_ends_at === null) {
-                            return Carbon::now()->addYear(500);
-                        } else {
-                            return $model->cfp_ends_at->addYear(1000);
-                        }
-                    });
+            case 'opening_next':
+                // Force CFPs with no CFP start date to the end
+                $conferences = $conferences->sortBy(function ($conference) {
```

Comment:
My comment about pushing to the end was about this one (opening_next), not the one you made the change on (date)

Reply:
I just removed the modification to the date sort and tested to ensure the opening_next sort sorts null cfp_ends_at below non-null cfp_ends_at.  There were no changes needed to be made to the switch case to get the functionality we wanted.